shopinvader/services/customer.py

```diff
@@ -68,37 +68,56 @@ def _validator_sign_in(self):
     def _validator_create(self):
         address = self.component(usage="addresses")
         schema = address._validator_create()
-        exclude_keys = ["type"]
-        for key in exclude_keys:
-            schema.pop(key, None)
         schema.update(
             {
-                "email": {"type": "string", "required": True},
-                "external_id": {"type": "string", "required": True},
-                "vat": {"type": "string", "required": False},
-                "company_name": {"type": "string", "required": False},
-                "function": {"type": "string", "required": False},
+                # Email is mandatory as of today
+                "email": {"type": "string", "required": True, "nullable": False},
             }
         )
+        for key in self._validator_create_exclude_keys():
+            schema.pop(key, None)
+        for key in self._validator_create_non_required_address_keys():
+            if key in schema:
+                schema[key]["required"] = False
+                schema[key]["nullable"] = True
         return schema
 
     def _validator_update(self):
         address = self.component(usage="addresses")
         schema = address._validator_update()
-        exclude_keys = ["type"]
-        for key in exclude_keys:
+        for key in self._validator_update_exclude_keys():
             schema.pop(key, None)
-        schema.update(
-            {
-                "email": {"type": "string", "required": False},
-                "external_id": {"type": "string", "required": False},
-                "vat": {"type": "string", "required": False},
-                "company_name": {"type": "string", "required": False},
-                "function": {"type": "string", "required": False},
-            }
-        )
+        for key in self._validator_update_non_required_address_keys():
+            if key in schema:
+                schema[key]["required"] = False
+                schema[key]["nullable"] = True
         return schema
 
+    def _validator_create_exclude_keys(self):
+        return ["type"]
+
+    def _validator_update_exclude_keys(self):
+        return self._validator_create_exclude_keys()
+
+    def _validator_create_non_required_address_keys(self):
+        # fmt: off
```

Comment:
What's this comment supposed to mean? 🤔

Reply:
skip black formatting to keep the list as is, which is more readable :wink: